src/client/pages/ConnectPayment/components/AdyenCheckout.tsx

```diff
@@ -143,9 +143,31 @@ export const AdyenCheckout: React.FC<Props> = ({ onSuccess }) => {
       history,
       onSuccess,
     }).mount(adyenCheckoutRef.current)
-  }, [paymentMethodsResponse, adyenLoaded])
+  }, [
+    currentLocale,
+    paymentMethodsResponse,
+    tokenizePaymentMutation,
+    submitAdditionalPaymentDetails,
+    history,
+    onSuccess,
```

Comment:
i'm unsure how this'll work. Since `onSuccess` for example probably will be a new function on every render idk how we should  treat that. Because `useEffect` checks object identity as far as i can remember this'll probably be triggered on every render of the parent component then, right? Meaning adyen will mount multiple times. That is probably not desired behaviour. Maybe that's resolvable but unless i remember the hooks api incorrectly this might have to be an exception to the rule, right?

Reply:
Since you feel unsure about this we should of course wait with doing anything with this component until I (or someone else) can actually test these changes. Thank you, that was my point with splitting the changes up in several commits and asking for a closer check on this particular file! 🙏